.changeset/engine-enhancements.md

```diff
@@ -0,0 +1,74 @@
+---
+"thirdweb": minor
+---
+
+Enhanced Engine functionality with search transactions and batch transaction support:
+
+- Added `Engine.searchTransactions()` to search for transactions by various filters (id, chainId, from address, etc.)
+
+  ```ts
+  // Search by transaction IDs
+  const transactions = await Engine.searchTransactions({
+    client,
+    filters: [
+      {
+        field: "id",
+        values: ["1", "2", "3"],
+        operation: "OR",
+      },
+    ],
+  });
+
+  // Search by chain ID and sender address
+  const transactions = await Engine.searchTransactions({
+    client,
+    filters: [
+      {
+        filters: [
+          {
+            field: "from",
+            values: ["0x1234567890123456789012345678901234567890"],
+            operation: "OR",
```

Comment:
do I *have* to define the operation even if I only pass one argument? also for `from` it could never be `AND` right? a tx can never be sent from multiple from addresses?

Reply:
Torn between keeping it the same as the API or wrapping into my own type / omitting stuff.

Could do some massaging on top but not sure if really needed